packages/e2e/lib/e2e.dart

```diff
@@ -78,33 +78,54 @@ class E2EWidgetsFlutterBinding extends LiveTestWidgetsFlutterBinding {
 
   static Map<String, String> _results = <String, String>{};
 
+  Map<String, dynamic> _data;
+
+  /// Add data to the reported result.
+  ///
+  /// The values in `data` should be json-serializable objects.
+  ///
+  /// For keys in `data` that's previously added, it will over write the
+  /// original values.
+  void addReportData(Map<String, dynamic> data) {
+    _data ??= <String, dynamic>{};
+    data.forEach((String key, dynamic value) {
+      _data[key] = value;
+    });
+  }
```

Comment:
This seems potentially confusing for users. Why not just expose the dictionary directly?

For example, how do I know if I'm overwriting a key? What if I wanted to keep the original value?

Reply:
I don't like exposing the dictionary for adding data because Dart doesn't provide an API to merge Map. If the way to write the data is through the dictionary itself, users will need to write an extra for loop to add things like a timeline (which is a `Map<String, dynamic>`). For me the most common use case would be to construct a json object and pass it to some method all at once. If over writing is the concern I would rather alway erase the data and rewrite everything, and expose only a getter.

Reply:
Or, to throw if there's key conflict, and add a `clearData` api to erase `_data`.

Reply:
With a double think I changed my mind.